ocaml-lsp-server/src/code_actions/action_remove_unused.ml

```diff
@@ -0,0 +1,81 @@
+open Import
+
+let action_title = "Remove unused"
+
+let slice doc (range : Range.t) =
+  let src = Document.source doc in
+  let (`Offset start) = Msource.get_offset src @@ Position.logical range.start
+  and (`Offset end_) = Msource.get_offset src @@ Position.logical range.end_ in
+  String.sub (Msource.text src) ~pos:start ~len:(end_ - start)
+
+let enclosing_value_binding_range pipeline pos name =
+  let open Option.O in
+  let typer = Mpipeline.typer_result pipeline in
+  let browse = Mbrowse.of_typedtree (Mtyper.get_typedtree typer) in
+  Mbrowse.enclosing
+    (Mpipeline.get_lexing_pos pipeline @@ Position.logical pos)
+    [ browse ]
+  |> List.find_map ~f:(function
+       | ( _
```

Comment:
Here we could take the second value of the tuple to make pattern matching more readable, ie
`List.find_map ~f:(fun n -> match snd n with ...`